host.go

```diff
@@ -650,6 +649,11 @@ func (h *gpbftHost) ReceiveDecision(decision *gpbft.Justification) (time.Time, e
 		log.Error(err)
 		return time.Time{}, err
 	}
+	if err := h.ec.Finalize(h.runningCtx, cert.ECChain.Head().Key); err != nil {
```

Comment:
I assume this can block if we're forced to sync to a chain we don't have? We should leave a comment if that's possible.

Reply:
How long would the sync take in practice? I wonder if it's worth optimising for some non-blocking sync then checkopint, and what the likelihood of the need for sync is to begin with.

Reply:
We often won't need to sync, but it can take an arbitrary amount of time (e.g., if we're really behind but we get a recent F3 decision).